datahub-graphql-core/src/main/resources/gms.graphql

```diff
@@ -1318,4 +1327,405 @@ enum ChartQueryType {
+    environment: [String]
+}
+
+type QuantitativeAnalyses {
+    """
+    Link to a dashboard with results showing how the model performed with respect to each factor
+    """
+    unitaryResults: ResultsType
+
+    """
+    Link to a dashboard with results showing how the model performed with respect to the intersection of evaluated factors?
+    """
+    intersectionalResults: ResultsType
+}
+
+union ResultsType = StringBox
+
+type CaveatsAndRecommendations {
+
+    """
+    Caveats on using this MLModel
+    """
+    caveats: CaveatDetails
+
+    """
+    Recommendations on where this MLModel should be used.
+    """
+    recommendations: String
+
+    """
+    Ideal characteristics of an evaluation dataset for this MLModel
+    """
+    idealDatasetCharacteristics: [String]
+}
+
+type CaveatDetails {
+
+    """
+    Did the results suggest any further testing?
+    """
+    needsFurtherTesting: Boolean
+
+    """
+    Caveat Description
+    """
+    caveatDescription: String
+
+    """
+    Relevant groups that were not represented in the evaluation dataset?
+    """
+    groupsNotRepresented: [String]
+}
+
+type EthicalConsiderations {
+
+    """
+    Does the model use any sensitive data (e.g., protected classes)?
+    """
+    data: [String]
+
+    """
+    Is the model intended to inform decisions about matters central to human life or flourishing – e.g., health or safety? Or could it be used in such a way?
+    """
+    humanLife: [String]
+
+    """
+    What risk mitigation strategies were used during model development?
+    """
+    mitigations: [String]
+
+    """
+    What risks may be present in model usage? Try to identify the potential recipients, likelihood, and magnitude of harms. If these cannot be determined, note that they were considered but remain unknown.
+    """
+    risksAndHarms: [String]
+
+    """
+    Are there any known model use cases that are especially fraught? This may connect directly to the intended use section
+    """
+    useCases: [String]
+}
+
+type BaseData {
+
+    """
+    Dataset used for the Training/Evaluation of the MLModel
+    """
+    dataset: String
+
+    """
+    Motivation to pick these datasets
+    """
+    motivation: String
+
+    """
+    Details of Data Proprocessing
+    """
+    preProcessing: [String]
+}
+
+type Metrics {
+
+    """
+    Measures of MLModel performance
+    """
+    performanceMeasures: [String]
+
+    """
+    Decision Thresholds used (if any)?
+    """
+    decisionThreshold: [String]
+
+}
+
+type IntendedUse {
+
+    """
+    Primary Use cases for the model.
+    """
+    primaryUses: [String]
+
+    """
+    Primary Intended Users
+    """
+    primaryUsers: [IntendedUserType]
+
+    """
+    Out of scope uses of the MLModel
+    """
+    outOfScopeUses: [String]
+}
+
+enum IntendedUserType {
+
+    ENTERPRISE
+
+    HOBBY
+
+    ENTERTAINMENT
+}
+
+type SourceCode {
+    """
+    Source Code along with types
+    """
+    sourceCode: [SourceCodeUrl]
```

Comment:
[SourceCodeUrl!] - --- Should this field be optional?

Reply:
No, its not. Added !